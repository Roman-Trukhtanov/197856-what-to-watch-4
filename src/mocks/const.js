export const MAX_GENRES_AMOUNT = 10;

export const MAX_VISIBLE_MOVIES_COUNT = 8;

export const OVERVIEW_STARRING_COUNT = 4;

export const ScreenType = {
  MAIN: `main`,
  MOVIE: `movie`,
};

export const GenreType = {
  ALL_GENRES: `All Genres`,
  COMEDIES: `Comedies`,
  CRIME: `Crime`,
  DOCUMENTARY: `Documentary`,
  DRAMAS: `Dramas`,
  THRILLERS: `Thrillers`,
  HORROR: `Horror`,
  ROMANCE: `Romance`,
  FAMILY: `Family`,
  FANTASY: `Fantasy`,
};

export const TabType = {
  OVERVIEW: `overview`,
  DETAILS: `details`,
  REVIEWS: `reviews`,
};
