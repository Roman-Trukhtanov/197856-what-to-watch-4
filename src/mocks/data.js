export const ScreenType = {
  MAIN: `main`,
  MOVIE: `movie`,
};
