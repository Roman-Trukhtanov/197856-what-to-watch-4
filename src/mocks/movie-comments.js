export const movieComments = [
  {
    id: 1,
    rating: 8.9,
    user: {
      id: 1,
      name: `Kate Muir`,
    },
    comment: `Discerning travellers and Wes Anderson fans will luxuriate in the glorious Mittel-European kitsch of one of the director's funniest and most exquisitely designed movies in years.`,
    dateText: `December 24, 2016`,
  },
  {
    id: 2,
    rating: 7.2,
    user: {
      id: 2,
      name: `Matthew Lickona`,
    },
    comment: `The mannered, madcap proceedings are often delightful, occasionally silly, and here and there, gruesome and/or heartbreaking.`,
    dateText: `December 20, 2016`,
  },
  {
    id: 3,
    rating: 8.0,
    user: {
      id: 3,
      name: `Bill Goodykoontz`,
    },
    comment: `Anderson's films are too precious for some, but for those of us willing to lose ourselves in them, they're a delight. "The Grand Budapest Hotel" is no different, except that he has added a hint of gravitas to the mix, improving the recipe.`,
    dateText: `November 18, 2015`,
  },
];
