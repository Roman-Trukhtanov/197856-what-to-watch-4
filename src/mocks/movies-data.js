export const TabType = {
  OVERVIEW: `overview`,
  DETAILS: `details`,
  REVIEWS: `reviews`,
};

export const tabs = [`overview`, `details`, `reviews`];

export const movieInfo = [
  {
    id: 0,
    title: `The Grand Budapest Hotel`,
    genre: `drama`,
    releaseYear: 2014,
    coverSrc: `/img/bg-the-grand-budapest-hotel.jpg`,
    bigPosterSrc: `img/the-grand-budapest-hotel-poster.jpg`,
  }
];

export const moviesOverview = [
  {
    id: 0,
    rate: 8.9,
    level: `Very good`,
    ratingCount: 240,
    description: [
      `In the 1930s, the Grand Budapest Hotel is a popular European ski resort, presided over by concierge Gustave H. (Ralph Fiennes). Zero, a junior lobby boy, becomes Gustave's friend and protege.`,
      `Gustave prides himself on providing first-class service to the hotel's guests, including satisfying the sexual needs of the many elderly women who stay there. When one of Gustave's lovers dies mysteriously, Gustave finds himself the recipient of a priceless painting and the chief suspect in her murder.`
    ],
    director: `Wes Andreson`,
    starring: `Bill Murray, Edward Norton, Jude Law, Willem Dafoe and other`,
  }
];

export const moviesDetails = [
  {
    id: 0,
    genre: `Comedy`,
    director: `Wes Andreson`,
    releaseYear: 2014,
    starring: [
      `Bill Murray`,
      `Edward Norton`,
      `Jude Law`,
      `Willem Dafoe`,
      `Saoirse Ronan`,
      `Tony Revoloru`,
      `Tilda Swinton`,
      `Tom Wilkinson`,
      `Owen Wilkinson`,
      `Adrien Brody`,
      `Ralph Fiennes`,
      `Jeff Goldblum`
    ],
    runTime: `1h 39m`,
  }
];

export const moviesComments = [
  {
    id: 0,
    comments: [
      {
        rate: 8.9,
        author: `Kate Muir`,
        description: `Discerning travellers and Wes Anderson fans will luxuriate in the glorious Mittel-European kitsch of one of the director's funniest and most exquisitely designed movies in years.`,
        date: `December 24, 2016`,
      },
      {
        rate: 7.2,
        author: `Matthew Lickona`,
        description: `The mannered, madcap proceedings are often delightful, occasionally silly, and here and there, gruesome and/or heartbreaking.`,
        date: `December 20, 2016`,
      },
      {
        rate: 8.0,
        author: `Bill Goodykoontz`,
        description: `Anderson's films are too precious for some, but for those of us willing to lose ourselves in them, they're a delight. "The Grand Budapest Hotel" is no different, except that he has added a hint of gravitas to the mix, improving the recipe.`,
        date: `November 18, 2015`,
      },
      {
        rate: 7.6,
        author: `Paula Fleri-Soler`,
        description: `It is certainly a magical and childlike way of storytelling, even if the content is a little more adult.`,
        date: `December 20, 2016`,
      },
      {
        rate: 8.0,
        author: `Amanda Greever`,
        description: `I didn't find it amusing, and while I can appreciate the creativity, it's an hour and 40 minutes I wish I could take back.`,
        date: `November 18, 2015`,
      },
      {
        rate: 7.0,
        author: `Paula Fleri-Soler`,
        description: `It is certainly a magical and childlike way of storytelling, even if the content is a little more adult.`,
        date: `December 20, 2016`,
      },
    ]
  }
];
