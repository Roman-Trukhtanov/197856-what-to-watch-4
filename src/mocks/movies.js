const PREVIEW_URL = `http://placeimg.com/280/175/any`;
const BUCK_TRAILER_SRC = `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`;
const BUNNY_TRAILER_SRC = `https://www.radiantmediaplayer.com/media/big-buck-bunny-360p.mp4`;

export default [
  {
    id: 0,
    title: `Кремниевая долина`,
    preview: `${PREVIEW_URL}/1`,
    videoSrc: BUCK_TRAILER_SRC,
    videoType: `video/mp4`,
  },
  {
    id: 1,
    title: `Форсаж`,
    preview: `${PREVIEW_URL}/2`,
    videoSrc: BUNNY_TRAILER_SRC,
    videoType: `video/webm`,
  },
  {
    id: 2,
    title: `Жажда скорости`,
    preview: `${PREVIEW_URL}/3`,
    videoSrc: BUCK_TRAILER_SRC,
    videoType: `video/mp4`,
  },
  {
    id: 3,
    title: `Гарри Поттер`,
    preview: `${PREVIEW_URL}/4`,
    videoSrc: BUNNY_TRAILER_SRC,
    videoType: `video/webm`,
  },
  {
    id: 4,
    title: `Спецназ`,
    preview: `${PREVIEW_URL}/5`,
    videoSrc: BUCK_TRAILER_SRC,
    videoType: `video/mp4`,
  },
  {
    id: 5,
    title: `Игра престолов`,
    preview: `${PREVIEW_URL}/6`,
    videoSrc: BUNNY_TRAILER_SRC,
    videoType: `video/webm`,
  },
  {
    id: 6,
    title: `Стрела`,
    preview: `${PREVIEW_URL}/7`,
    videoSrc: BUCK_TRAILER_SRC,
    videoType: `video/mp4`,
  },
  {
    id: 7,
    title: `Флэш`,
    preview: `${PREVIEW_URL}/8`,
    videoSrc: BUNNY_TRAILER_SRC,
    videoType: `video/webm`,
  },
];
