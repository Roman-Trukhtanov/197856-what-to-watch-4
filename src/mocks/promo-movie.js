export default {
  title: `The Grand Budapest Hotel`,
  genre: `drama`,
  year: 2014
};
