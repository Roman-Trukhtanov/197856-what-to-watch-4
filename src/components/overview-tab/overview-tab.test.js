import React from "react";
import renderer from "react-test-renderer";
import OverviewTab from "./overview-tab";
import {moviesOverview} from "../../mocks/movies-data";

describe(`OverviewTab component`, () => {
  it(`Render OverviewTab`, () => {
    const tree = renderer
      .create(<OverviewTab
        movieOverview={moviesOverview[0]}
      />)
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
