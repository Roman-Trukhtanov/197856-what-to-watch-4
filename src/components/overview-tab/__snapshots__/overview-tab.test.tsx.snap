// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OverviewTab component Render OverviewTab 1`] = `
Array [
  <div
    className="movie-rating"
  >
    <div
      className="movie-rating__score"
    >
      3.0
    </div>
    <p
      className="movie-rating__meta"
    >
      <span
        className="movie-rating__level"
      >
        Bad
      </span>
      <span
        className="movie-rating__count"
      >
        100 ratings
      </span>
    </p>
  </div>,
  <div
    className="movie-card__text"
  >
    <p>
      Description
    </p>
    <p
      className="movie-card__director"
    >
      <strong>
        Director: Guy Ritchie
      </strong>
    </p>
    <p
      className="movie-card__starring"
    >
      <strong>
        Starring: Jason Statham, Brad Pitt, Benicio Del Toro
      </strong>
    </p>
  </div>,
]
`;
